src/btree/bt_split.c

```diff
@@ -773,7 +773,8 @@ __split_multi_inmem(
 	 * transaction to an impossibly old value so this page is never skipped
 	 * in a checkpoint.
 	 */
-	page->modify->first_dirty_txn = WT_TXN_FIRST;
+	if (page->modify != NULL)
+		page->modify->first_dirty_txn = WT_TXN_FIRST;
```

Comment:
The modify structure can be NULL, because in this brave new world we may be creating clean pages when doing an in-memory split. It used to be the case that we only re-instantiated pages when updates couldn't be evicted, a ``WT_EVICT_IN_MEMORY`` split potentially (hopefully) creates a number of clean pages.

I'll try to get that context into a comment.